zkevm-circuits/src/evm_circuit/execution/jumpi.rs

```diff
@@ -47,18 +46,25 @@ impl<F: Field> ExecutionGadget<F> for JumpiGadget<F> {
 
         // Lookup opcode at destination when should_jump
         cb.condition(should_jump.clone(), |cb| {
-            cb.opcode_lookup_at(
-                from_bytes::expr(&destination.cells),
-                OpcodeId::JUMPDEST.expr(),
+            cb.require_equal(
+                "JUMPI destination must be within range if condition is non-zero",
+                dest.within_range(),
                 1.expr(),
             );
+
+            cb.opcode_lookup_at(dest.valid_value(), OpcodeId::JUMPDEST.expr(), 1.expr());
         });
 
+        cb.require_zero(
+            "JUMPI condition must be 0 if destination is Uint64 overflow",
+            dest.overflow() * not::expr(is_condition_zero.expr()),
+        );
+
```

Comment:
I think this constraint is redundant with `"JUMPI destination must be within range if condition is non-zero"`:
First one is:
- should_jump == 1 -> dest.within_range == 1
Second one is:
- dest.overflow == 1 -> not(is_condition_zero) == 0

And we have:
- should_jump = not(is_condition_zero)
- dest.within_range = not(dest.overflow)

So the second one can be rewritten as:
- dest.within_range == 0 -> should_jump == 0
which is:
- not(dest.within_range == 1) -> not(should_jump == 1)

And this is equivalent to the first one following the contraposition rule https://en.wikipedia.org/wiki/Contraposition

Let me know if I missed something in my reasoning.

Reply:
Delete this redundant constraint in commit [787093f](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1317/commits/787093ffc611c084e83d786ff64245768163aff3).